### PR TITLE
color-code Proficiencies that cannot be learned in the recipe screen

### PR DESCRIPTION
**Pull request by Jamuro-g**
#### Summary
SUMMARY: Interface "color-code Proficiencies that cannot be learned in the recipe screen"

#### Purpose of change
Currently there is no ui indication whether a player is eligible to gain experience for an advanced proficiency in the recipe screen.
This change marks the names of proficiencies that have missing requirements as red.
-->

#### Describe the solution
A recipes missing proficiencies are checked for prerequisites and marked red if the player hasn't already mastered the needed proficiency to unlock xp gain.

#### Describe alternatives you've considered
Idealy, the recipe ui would not only mark proficiencies, but also list the requirements in a non obtrusive way that doesn't add to the clutter. Tooltips as a functionality come to mind, but that would exceed my experience with the project at this time.

#### Testing
Create a character, debug skills in and check recipe ui for the different prof colorations. (cyan = default, red = cannot be learned yet)

### CODE REVIEW

**Comment by BrettDong** [comment]
You added some irrelevant files to git.

**Comment by BrettDong** [comment]
Please run `make astyle` on your local machine to format source code before making git commit.

**Comment by BrettDong** [comment]
I recommend you to remove your IDE files on your local machine, amend the commit and force-push to remote.

**Comment by Jamuro-g** [comment]
> I recommend you to remove your IDE files from git on your local machine, amend the commit and force-push to remote.

yeah, sorry i ll try and clean this up.

File: src/recipe.cpp [hunk 1]
```diff
@@ -680,9 +681,13 @@ std::string recipe::missing_proficiencies_string( Character *c ) const
...
 
     std::string color = "yellow";
+    std::string name_color = "cyan";
     std::string missing = enumerate_as_string( missing_profs.begin(),
     missing_profs.end(), [&]( const prof_penalty & prof ) {
-        return profstring( prof, color );
+        if( !c->has_prof_prereqs( prof.id ) ) {
+            name_color = "red";
+        }
+        return profstring( prof, color, name_color );
```
Comment (BrettDong) [inline src/recipe.cpp:690]: This makes `name_color` red for all subsequent proficiencies regardless of mastered or not. Did you want to express
```suggestion
        return profstring( prof, color, c->has_prof_prereqs( prof.id ) ? "cyan" : "red" );
```

Comment (Jamuro-g) [reply src/recipe.cpp]: thank you, should be fixed now.

File: src/recipe.cpp [hunk 2]
```diff
@@ -682,7 +683,7 @@ std::string recipe::missing_proficiencies_string( Character *c ) const
     std::string color = "yellow";
     std::string missing = enumerate_as_string( missing_profs.begin(),
     missing_profs.end(), [&]( const prof_penalty & prof ) {
-        return profstring( prof, color );
+        return profstring(prof, color, c->has_prof_prereqs(prof.id) ? "cyan" : "red");
```
Comment (BrettDong) [inline src/recipe.cpp:686]:
```suggestion
        return profstring( prof, color, c->has_prof_prereqs( prof.id ) ? "cyan" : "red" );
```

File: src/recipe.cpp [hunk 3]
```diff
@@ -609,24 +609,25 @@ struct prof_penalty {
...
-                              prof.id->name(), color, prof.time_mult, mitigated_str );
+        return string_format( _( "<color_%s>%s</color> (<color_%s>%gx\u00a0time</color>%s)" ),
+                              name_color, prof.id->name(), color, prof.time_mult, mitigated_str );
     }
 
     return string_format(
-               _( "<color_cyan>%s</color> (<color_%s>%gx\u00a0time, %gx\u00a0failure</color>%s)" ),
-               prof.id->name(), color, prof.time_mult, prof.failure_mult, mitigated_str );
+               _( "<color_%s>%s</color> (<color_%s>%gx\u00a0time, %gx\u00a0failure</color>%s)" ),
+               name_color, prof.id->name(), name_color, prof.time_mult, prof.failure_mult, mitigated_str );
```
Comment (BrettDong) [inline src/recipe.cpp:630]: Is it intentional to also change colour of multipliers? I notice that _Principles of Metalworking (1.5x time, 5x failure)_ is no longer in yellow.
unpatched
(unpatched)

patched
(patched)

Comment (Jamuro-g) [reply src/recipe.cpp:630]: i ll take a look at it. the goal was to only change the color of the advanced profs you can't learn and not all of them.

**Review by BrettDong — APPROVED** [review]

**Pull request merged**